### DISCUSSION TITLE
How can i get param in route.ts file in nextjs13 app directory

### DISCUSSION THREAD

**Discussion by zwelhtetyan** [vercel/next.js#51486] [App Router] [answered]
app
--api
---menus
----[menuName]
-----route.ts

if i request with "http:localhost:3000/api/menus/`pizza`" , I want to get `pizza` dynamically.

**Comment by icyJoseph** [comment, marked as answer]
The route handlers take in a second argument. Please refer to https://nextjs.org/docs/app/building-your-application/routing/router-handlers#dynamic-route-segments this section of the docs.

### Code sample

```ts
export async function GET(
  request: Request,
  { params }: { params: { slug: string } }
) {
  const slug = params.slug // 'a', 'b', or 'c'
}
```

### Expectations chart



Route | Example URL | params
-- | -- | --
app/items/[slug]/route.js | /items/a | { slug: 'a' }
app/items/[slug]/route.js | /items/b | { slug: 'b' }
app/items/[slug]/route.js | /items/c | { slug: 'c' }

<br class="Apple-interchange-newline">

Route | Example URL | params
-- | -- | --
app/items/[slug]/route.js | /items/a | { slug: 'a' }
app/items/[slug]/route.js | /items/b | { slug: 'b' }
app/items/[slug]/route.js | /items/c | { slug: 'c' }

<br class="Apple-interchange-newline">

Route | Example URL | params
-- | -- | --
app/items/[slug]/route.js | /items/a | { slug: 'a' }
app/items/[slug]/route.js | /items/b | { slug: 'b' }
app/items/[slug]/route.js | /items/c | { slug: 'c' }

<br class="Apple-interchange-newline">

Route | Example URL | params
-- | -- | --
app/items/[slug]/route.js | /items/a | { slug: 'a' }
app/items/[slug]/route.js | /items/b | { slug: 'b' }
app/items/[slug]/route.js | /items/c | { slug: 'c' }

<br class="Apple-interchange-newline">

**Reply by zwelhtetyan** [reply]
It works, Thanks a lot @icyJoseph for your help ❤️
